engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java

```diff
@@ -11,27 +11,62 @@
 import org.hibernate.search.engine.cfg.BackendSettings;
 import org.hibernate.search.engine.cfg.ConfigurationPropertySource;
 import org.hibernate.search.engine.cfg.EngineSettings;
+import org.hibernate.search.engine.cfg.spi.ConfigurationPropertySourceScopeUtils;
+import org.hibernate.search.engine.environment.bean.BeanResolver;
 
 public final class EngineConfigurationUtils {
 
 	private EngineConfigurationUtils() {
 	}
 
-	public static ConfigurationPropertySourceExtractor extractorForBackend(Optional<String> backendNameOptional) {
+	public static ConfigurationPropertySourceExtractor extractorForBackend(BeanResolver beanResolver,
+			Optional<String> backendNameOptional) {
 		if ( !backendNameOptional.isPresent() ) {
-			return engineSource -> engineSource.withMask( EngineSettings.Radicals.BACKEND );
+			return engineSource -> engineSource.withMask( EngineSettings.Radicals.BACKEND )
+					.withFallback(
+							ConfigurationPropertySourceScopeUtils.fallback(
+									beanResolver,
+									ConfigurationPropertySourceScopeUtils.backend()
+							)
+					);
 		}
 		else {
 			return engineSource -> engineSource.withMask( EngineSettings.Radicals.BACKENDS )
-					.withMask( backendNameOptional.get() );
+					.withFallback(
+							ConfigurationPropertySourceScopeUtils.fallback(
+									beanResolver,
+									ConfigurationPropertySourceScopeUtils.backend()
+							)
+					)
+					.withMask( backendNameOptional.get() )
+					.withFallback(
+							ConfigurationPropertySourceScopeUtils.fallback(
+									beanResolver,
+									ConfigurationPropertySourceScopeUtils.backend( backendNameOptional.get() )
+							)
+					);
 		}
 	}
 
 	public static ConfigurationPropertySourceExtractor extractorForIndex(
-			ConfigurationPropertySourceExtractor extractorForBackend, String indexName) {
+			BeanResolver beanResolver, ConfigurationPropertySourceExtractor extractorForBackend,
+			String backendName, String indexName) {
 		return engineSource -> {
 			ConfigurationPropertySource backendSource = extractorForBackend.extract( engineSource );
-			return backendSource.withMask( BackendSettings.INDEXES ).withMask( indexName )
+			return backendSource.withMask( BackendSettings.INDEXES )
+					.withFallback(
+							ConfigurationPropertySourceScopeUtils.fallback(
+									beanResolver,
+									ConfigurationPropertySourceScopeUtils.index( backendName )
+							)
+					)
```

Comment:
I've described it here -https://github.com/hibernate/hibernate-search/pull/3632/files#diff-ad0db95d3e615f58b4b8df315613a90e39e11f911c95fb6f9acd4ec1db4d917bR22 . 

> There is no notion of default index

While that's true, the idea was to be a bit more generic and not tie things too much to the specifics of a particular scope. But I suppose since we are going with a simpler approach overall... I'll remove that and change the docs.